kornia/geometry/transform/homography_warper.py

```diff
@@ -23,6 +25,117 @@
+
+    Example:
+       >>> img = torch.rand(1, 4, 5, 6)
+       >>> H = torch.eye(3)[None]
+       >>> out = warp_perspective(img, H, (4, 2), align_corners=True)
+       >>> print(out.shape)
+       torch.Size([1, 4, 4, 2])
+
+    .. note::
+        This function is often used in conjuntion with :func:`get_perspective_transform`.
+
+    .. note::
+        See a working example `here <https://kornia.readthedocs.io/en/latest/
+        tutorials/warp_perspective.html>`_.
+    """
+    if not isinstance(src, torch.Tensor):
+        raise TypeError("Input src type is not a torch.Tensor. Got {}".format(type(src)))
+
+    if not isinstance(M, torch.Tensor):
+        raise TypeError("Input M type is not a torch.Tensor. Got {}".format(type(M)))
+
+    if not len(src.shape) == 4:
+        raise ValueError("Input src must be a BxCxHxW tensor. Got {}".format(src.shape))
+
+    if not (len(M.shape) == 3 and M.shape[-2:] == (3, 3)):
+        raise ValueError("Input M must be a Bx3x3 tensor. Got {}".format(M.shape))
+
+    if not normalized_homography:
+        # TODO: remove the statement below in kornia v0.6
+        if align_corners is None:
+            message: str = (
+                "The align_corners default value has been changed. By default now is set True "
+                "in order to match cv2.warpPerspective. In case you want to keep your previous "
+                "behaviour set it to False. This warning will disappear in kornia > v0.6."
+            )
+            warnings.warn(message)
+            # set default value for align corners
+            align_corners = True
+
+        B, C, H, W = src.size()
+        h_out, w_out = dsize
+
+        # we normalize the 3x3 transformation matrix and convert to 3x4
+        dst_norm_trans_src_norm: torch.Tensor = normalize_homography(M, (H, W), (h_out, w_out))  # Bx3x3
+
+        src_norm_trans_dst_norm = _torch_inverse_cast(dst_norm_trans_src_norm)  # Bx3x3
+
+        # this piece of code substitutes F.affine_grid since it does not support 3x3
+        grid = (
+            create_meshgrid(h_out, w_out, normalized_coordinates=normalized_coordinates, device=src.device)
+            .to(src.dtype)
+            .repeat(B, 1, 1, 1)
+        )
+        grid = transform_points(src_norm_trans_dst_norm[:, None, None], grid)
+
+    else:
+        if align_corners is None:
+            align_corners = False
+
+        if not M.device == src.device:
+            raise TypeError(
+                "Patch and homography must be on the same device. \
+                            Got patch.device: {} M.device: {}.".format(
+                    src.device, M.device
+                )
+            )
+
+        height, width = dsize
+        grid = create_meshgrid(height, width, normalized_coordinates=normalized_coordinates)
+        grid = warp_grid(grid, M)
+
+    return F.grid_sample(src, grid, mode=mode, padding_mode=padding_mode, align_corners=align_corners)
```

Comment:
ideally, to encapsulate things we could probably call `remap` here - however we might need to unbind -> remap -> stack the grid which can be slightly inefficient. One idea, is to have an overloaded `remap_grid` that directly takes the `BxHxWx2` grid. /cc @ducha-aiki @lferraz